packages/thirdweb/src/utils/units.ts

```diff
@@ -125,10 +127,22 @@
  * toWei('1')
  * // 1000000000000000000n
  * ```
+ *
+ * For chains with varying decimals, you can provide the chain.
+ * ```ts
+ * import { toWei } from "thirdweb/utils";
+ * toWei('1', defineChain(295))
+ * // 10000000n
+ * ```
  * @utils
  */
-export function toWei(tokens: string) {
-  return toUnits(tokens, 18);
+export function toWei(tokens: string, chain?: Chain) {
```

Comment:
wonder if a better call would be to get the chain metadata async in the places where we do toWei above, and use toUnits with the chain decimals?

Reply:
It's not a native currency decimals thing, it's the literal wei on the chain. Peep the slack thread linked in the PR description

Reply:
Yeah but I assume we put this decimals value in the chains db right?

Reply:
We don't as of right now because (based on my understanding) it's something EVM chains would normally never deviate from. We have the native currency decimals, but that's different. It's an extreme edge case as Firekeeper said

Reply:
gotcha

Reply:
I agree, I hate it but not sure of any other solution